implementation/src/main/java/io/smallrye/config/mapper/ConfigMapping.java

```diff
@@ -636,6 +639,16 @@ public Result mapConfiguration(SmallRyeConfig config) throws ConfigurationValida
         return new Result(context.getRootsMap());
     }
 
+    private boolean isPropertyInRoot(String propertyName) {
+        final Set<String> registeredRoots = roots.keySet();
+        for (String registeredRoot : registeredRoots) {
+            if (propertyName.startsWith(registeredRoot)) {
```

Comment:
A prefix check is not sufficient here.  Roots might not cover a whole namespace - for example, in some use cases, it might be necessary to match a property name that is outside of the root in a space that isn't directly controlled by the author.  It should be a pattern which can be built into a `KeyMap`, and specified on the root as an optional configuration (e.g. `quarkus.**`).

Reply:
We can enhance this. The issue here was about properties that may not be mapped (and you don't want to map) for instance, all the properties that you get com System and Env that you normally don't use.